api/src/main/java/org/eclipse/microprofile/concurrent/spi/ThreadContextProvider.java

```diff
@@ -0,0 +1,143 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.concurrent.spi;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * <p>Third party providers of thread context implement this interface to enable the
+ * provided type of context to participate in thread context capture & propagation
+ * when the MicroProfile Concurrency specification's <code>ManagedExecutor</code>
+ * and <code>ThreadContext</code> are used to create and contextualize dependent
+ * actions and tasks.</p>
+ *
+ * <p>Application code must never access the classes within this <code>spi</code>
+ * package. Instead, application code uses the <code>ManagedExecutor</code> and
+ * <code>ThreadContext</code> interfaces.</p>
+ *
+ * <p>The <code>ThreadContextProvider</code> implementation and related classes are
+ * packaged within the third party provider's JAR file. The implementation is made
+ * discoverable via the standard <code>ServiceLoader</code> mechanism. The JAR file
+ * that packages it must include a file of the following name and location,</p>
+ *
+ * <code>META-INF/services/org.eclipse.microprofile.concurrent.spi.ThreadContextProvider</code>
+ *
+ * <p>The content of the aforementioned file must be a single line that specifies the
+ * fully qualified name of the <code>ThreadContextProvider</code> implementation.</p>
+ *
+ * <p><code>ManagedExecutor</code> and <code>ThreadContext</code> must use the
+ * <code>ServiceLoader</code> to identify all available implementations of
+ * <code>ThreadContextProvider</code> that can participate in thread context capture
+ * & propagation and must invoke them either to capture current thread context or establish
+ * default thread context per the configuration of the <code>ManagedExecutor</code> or
+ * <code>ThreadContext</code> instance wherever these interfaces are used to create
+ * and contextualize dependent actions and tasks.</p>
+ */
+public interface ThreadContextProvider {
+    /**
+     * Captures from the current thread a snapshot of the provided thread context type.
+     *
+     * @param props provided for compatibility with EE Concurrency spec, which allows
+     *        for specifying a set of 'execution properties' when capturing thread context.
+     *        Thread context providers that don't supply or use execution properties
+     *        can ignore this parameter.
+     * @return immutable snapshot of the provided type of context, captured from the
+     *         current thread. NULL must be returned if the thread context provider
+     *         does not support capturing context from the current thread and
+     *         propagating it to other threads.
+     */
+    ThreadContextSnapshot currentContext(Map<String, String> props);
```

Comment:
Should this be String, String? Maybe String, Object has more coverage.

Reply:
It is `Map<String, String>` for compatibility with EE Concurrency execution properties. Switching to Object would force code to deal with unwanted casting.